testool/src/statetest/executor.rs

```diff
@@ -350,7 +339,12 @@ pub fn run_test(
         builder = _builder;
 
         let prover = MockProver::run(k, &circuit, instance).unwrap();
-        prover.assert_satisfied_par();
+        prover
+            .verify()
```

Comment:
```suggestion
            .verify_par()
```
Suggestion to get parallelized MockProver verification

Reply:
@adria0 Before working on this PR, I thought that `assert_satisfied_par` dumps richer output, too.

However, the [report](http://testool-public.s3-website.eu-central-1.amazonaws.com/nightly.1695857921.ee97ab1.html#all) shows only "circuit was not satisfied" but not the rich output. My theory is that `assert_satisfied_par` panicked so the output was not captured by `StateTestError`, that's why I customized a StateTestError to capture the VerifyFailure.